include/tgfx/core/Stream.h

```diff
@@ -65,4 +66,28 @@ class Stream {
    */
   virtual bool rewind() = 0;
 };
+
+/**
+ * StreamFactory creates instances of Stream and can be inherited externally to implement the
+ * loading of cache files without local path, such as caching data into a database, or assets
+ * resource files on the Android platform.
+ */
+class StreamFactory {
+ public:
+  /**
+  * Registers custom protocols with the specified factory. The factory will create streams for
+  * paths that start with the specified protocol. Custom protocol can use protocol headers like
+  * "assets://".
+  */
```

Comment:
Registers custom protocols with the specified factory. The factory will create streams for paths that start with the specified protocol, such as "assets://".